script/scripts/CompoundingMigration.s.sol

```diff
@@ -0,0 +1,166 @@
+import "../../src/zkbob/ZkBobDirectDepositQueue.sol";
+import "../../src/infra/UniswapV3Seller.sol";
+import "../../src/zkbob/ZkBobPoolUSDC.sol";
+import "@aave/aave-vault/src/ATokenVault.sol";
+import "../../src/zkbob/ZkBobPoolUSDCMigrated.sol";
+
+contract BOBPoolMigration is Script, StdCheats {
+    ZkBobPoolUSDCMigrated pool = ZkBobPoolUSDCMigrated(0x72e6B59D4a90ab232e55D4BB7ed2dD17494D62fB);
+    ZkBobDirectDepositQueue queue_proxy = ZkBobDirectDepositQueue(0x668c5286eAD26fAC5fa944887F9D2F20f7DDF289);
+    address usdc_addr = address(0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174);
+    address relayer_addr = address(0xc2c4AD59B78F4A0aFD0CDB8133E640Db08Fa5b90);
+    address poolAddressesProvider = address(0xa97684ead0e402dC232d5A977953DF7ECBaB3CDb);
+    uint256 D = 10 ** 12;
+    address yieldAddress;
+
+    struct VerificationValues {
+        uint256 withdrawalDiff;
+        uint256 depositDiff;
+        uint256 fees;
+        uint256 ddIn;
+        uint256 ddOutDiff;
+        ZkBobAccounting.Limits limits;
+    }
+
+    function migrate() internal {
+        ITransferVerifier transferVerifier = pool.transfer_verifier();
+        ITreeVerifier treeVerifier = pool.tree_verifier();
+        IBatchDepositVerifier batchDepositVerifier = pool.batch_deposit_verifier();
+        uint256 pool_id = pool.pool_id();
+        require(queue_proxy == pool.direct_deposit_queue(), "Incorrect Direct Deposit queue proxy");
+
+        vm.startPrank(deployer);
+        ZkBobPoolUSDCMigrated poolImpl = new ZkBobPoolUSDCMigrated(
+            pool_id, usdc_addr,
+            transferVerifier, treeVerifier, batchDepositVerifier,
+            address(queue_proxy)
+        );
+        UniswapV3Seller seller = new UniswapV3Seller(uniV3Router, uniV3Quoter, usdc_addr, 500, address(0), 0);
+        ZkBobDirectDepositQueue queueImpl = new ZkBobDirectDepositQueue(address(pool), usdc_addr, 1);
+        vm.stopPrank();
+
+        vm.startPrank(owner);
+        IERC20(usdc_addr).approve(address(pool), type(uint256).max);
+        EIP1967Proxy(payable(address(pool))).upgradeTo(address(poolImpl));
+        IERC20(usdc_addr).approve(address(pool), 0);
+        EIP1967Proxy(payable(address(queue_proxy))).upgradeTo(address(queueImpl));
+        pool.setTokenSeller(address(seller));
+
+        ATokenVault yieldVault = new ATokenVault(usdc_addr, 4546, IPoolAddressesProvider(poolAddressesProvider));
+        deal(usdc_addr, owner, 1_000_050 ether / D);
+        address yieldProxyAddr = computeCreateAddress(address(owner), vm.getNonce(address(owner)));
+        IERC20(usdc_addr).approve(yieldProxyAddr, type(uint256).max);
+        bytes memory initData = abi.encodeWithSelector(
+            ATokenVault.initialize.selector,
+            address(owner),
+            0.2 ether, // 20%
+            "Wrapped BOB Yield Token",
+            "wBYT",
+            1_000_000 ether / D
+        );
+        TUP yieldProxy = new TUP(address(yieldVault), owner, initData);
```

Comment:
Please research what could be a procedure if pure `EIP1967` proxy is used here instead of `TransparentUpgradeableProxy`. `TransparentUpgradeableProxy` uses [extra SLOAD for every `fallback` invocation](https://github.com/aave/Aave-Vault/blob/028d7696b323ae582dc7c43d3678789080a3ba92/lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol#L121).